c++/src/kj/filesystem-disk-test.c++

```diff
@@ -945,5 +946,30 @@ KJ_TEST("DiskFile holes") {
 }
 #endif
 
+// Ensure the current path is correctly computed.
+//
+// See issue #1425.
+KJ_TEST("DiskFilesystem::computeCurrentPath") {
+  TempDir tempDir;
+  auto dir = tempDir.get();
+
+  // Paths can be PATH_MAX, but the segments which make up that path typically
+  // can't exceed 255 bytes.
+  auto maxPathSegment = std::string(255, 'a');
+
+  // Create a path which exceeds the 256 byte buffer used in
+  // computeCurrentPath.
+  auto subdir = dir->openSubdir(Path({
+    maxPathSegment,
+    maxPathSegment,
+    "some_path_longer_than_256_bytes"
+  }), WriteMode::CREATE | WriteMode::CREATE_PARENT);
+
+  KJ_SYSCALL(fchdir(KJ_ASSERT_NONNULL(subdir->getFd())));
+
+  // Test computeCurrentPath indirectly.
+  KJ_EXPECT(newDiskFilesystem());
```

Comment:
KJ_EXPECT doesn't really make sense here since `newDiskFilesystem()` doesn't return a boolean. I think this only complies by accident.

Reply:
To be clear, you can just remove the `KJ_EXCEPT`. This function wont produce a warning if the result is unused. `fchdir` is apparently specially marked to produce such a warning.

Reply:
Done! Thank you again for being so helpful!